test_spack.py

```diff
@@ -0,0 +1,214 @@
+#!/usr/bin/python
+
+import unittest
+import sys
+import subprocess
+
+# Maps packages to Set[use cases].
+# The comment after each use case states why this use case exists or where it appears.
+use_cases = {
+    'atlas' : {},
```

Comment:
we could simplify the number of use cases by having only the top nodes of the tree? 
For example for the tree 'icondusk-e2e' : {'atlas_utilities', 'dawn4py', 'dusk', 'atlas', 'cuda'}, 
triggering dawn, will also automatically trigger the parent that depends on dawn, i.e. icondusk-e2e (via dawn4py). 
And icondusk-e2e will then traverse the tree down and compile all dependencies i.e. dawn. 
Therefore it seems redundant, and we could have only the top nodes of the tree ? 
What do you think ?

Reply:
I think this strategy could miss some use cases that are only triggered by lower-level packages. Say we introduce a new package that depends on cosmo, but only has a single use case (yet) with ~cppdycode. We would never test cosmo +cppdycore then, right?
Also, if the cache isn't wiped in between testing use cases, the true redundancies will be cached.